File: intentresolver/src/main/java/com/walhalla/intentresolver/FileView.java
package com.walhalla.intentresolver;

import android.content.Intent;

import java.io.File;

public interface FileView {
    void openFolderChooser(Intent intent);
    void showSelectedFolder(File folderPath);
    void showError(String message);

    //void shareMp4Selector(File file, String o);
}

File: intentresolver/src/main/java/com/walhalla/intentresolver/FileView.kt
package com.walhalla.intentresolver

import android.content.Intent
import java.io.File

interface FileView {
    fun openFolderChooser(intent: Intent?)
    fun showSelectedFolder(folderPath: File?)
    fun showError(message: String?) //void shareMp4Selector(File file, String o);
}